doc/patch-author-guide.md

```diff
@@ -113,8 +113,15 @@ patch module is loaded.
 
 `kpatch-macros.h` provides `KPATCH_LOAD_HOOK` and `KPATCH_UNLOAD_HOOK` macros
 to define such functions.  The signature of both hook functions is `void
-foo(void)` and and they may run in `stop_machine` context (so they must not
-sleep).
+foo(void)`.  Their execution context is as follows:
+
+* For objects that are already loaded, hook functions will be run by
```

Comment:
The term 'object' has a specific meaning in our code, but some readers of this doc might not grok what it means.  So maybe this should say something like "For patches to vmlinux or for kernel modules which have already been loaded ..."